src/nn/evaluate.c

```diff
@@ -16,7 +16,10 @@
 
 #include "evaluate.h"
 
+#if defined(__x86_64__) || defined(_M_X64)
 #include <immintrin.h>
+#endif

```

Comment:
Similar to above (but ignore the copy into every place part), I would use the same `defined()` logic that actually causes me to require this include.

Reply:
I am not sure I understand correctly. Placing the `defined()` in the proper places would mean to copy it all over indeed, as with accumulator.h file.

Or do you rather prefer:

`#if defined(__AVX512F__) || defined(__AVX512BW__) || defined(__AVX2__) || defined(__SSE2__)`
`#include <immintrin.h> `
`#endif`